pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -99,10 +99,10 @@ func (a *index) WorkloadsCollection(
 				return t.ResourceName()
 			}))
 			// We could do a non-FilterGeneric but krt currently blows up if we depend on the same collection twice
-			auths := fetchPeerAuthentications(ctx, PeerAuths, meshCfg, se.Namespace, p.Labels)
+			auths := fetchPeerAuthentications(ctx, PeerAuths, meshCfg, se.Namespace, wle.Labels)
 			policies = append(policies, convertedSelectorPeerAuthentications(meshCfg.GetRootNamespace(), auths)...)
 			var waypoint *Waypoint
-			if p.Labels[constants.ManagedGatewayLabel] != constants.ManagedGatewayMeshControllerLabel {
+			if wle.Labels[constants.ManagedGatewayLabel] != constants.ManagedGatewayMeshControllerLabel {
 				// Waypoints do not have waypoints, but anything else does
```

Comment:
What are we referring to when we say anything?

Reply:
This comments make no sense at this place. Actually it means service/pod/wle/namespace can have waypoint capturred